app/domain/authentication/authn_jenkins/authenticator.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+require 'json'
+
+module Authentication
+  module AuthnJenkins
+    class Authenticator
+
+      Err = Errors::Authentication::AuthnJenkins
+      def initialize(env:)
+        @env = env
+      end
+
+      def webservice
+        @webservice ||= ::Authentication::Webservice.new(
+          account:            @account,
+          authenticator_name: @authenticator_name,
+          service_id:         @service_id
+        )
+      end
+
+      def http_get_request(url, username, password) 
+        begin
+          uri = URI.parse(url)
+          http = Net::HTTP.new(uri.host, uri.port)
+          http.read_timeout = 5
+          request = Net::HTTP::Get.new(uri.request_uri)
+          request.basic_auth(username, password)
+          http.request(request)
+        rescue Net::OpenTimeout => e
+          Rails.logger.error("Timeout to Jenkins host Exception #{e}. Jenkins host '#{url}' is most likely invalid. Validate this url can be accessed from the conjur server.")
+          raise Err::HostNotFound, url
+        end
+      end
+
+      def get_build_info(jenkins_url, job_path, build_number, username, password)
+        jenkins_build_endpoint = "/job/#{job_path}/#{build_number}/api/json"
+        jenkins_build_url = "#{jenkins_url}#{jenkins_build_endpoint}"
+        http_get_request(jenkins_build_url, username, password)
+      end
+
+      def build_running?(response)
+        body = response.body
+        status_code = response.code
+
+        Rails.logger.debug("Jenkins Build Response: #{body}")
+        if status_code != '200'
+          raise Err::BuildInfoError.new(jenkinsBuildURL, "#{status_code} - #{body}")
```

Comment:
The variable `jenkinsBuildURL` does not exist in the scope of this method.  You'd get an exception trying to throw this exception :stuck_out_tongue_closed_eyes

Reply:
Removed jenkinsBuildURL from the exception in this commit: https://github.com/AndrewCopeland/conjur/commit/d16c40c290d4f1ac139b101cd089a1635f20db29